wally-registry-backend/src/main.rs

```diff
@@ -96,6 +96,32 @@ async fn publish(
     index.update()?;
 
     let manifest = get_manifest(&mut archive).status(Status::BadRequest)?;
+    let package_id = manifest.package_id();
+
+    let scope = package_id.name().scope();
+    let package_owners = index.get_scope_owners(&scope)?;
+
+    let has_authorization = match authorization.github() {
+        None => Ok(()), // We authenticated using another method
+        Some(github_info) => match package_owners {
+            None if github_info.login() == scope => {
+                index
+                    .add_scope_owner(scope, github_info.id())
+                    .context("Could not add owner to scope")?;
+                Ok(())
+            }
+            None => Err("you cannot claim this scope"),
+            Some(owners) => match owners.iter().any(|owner| owner == github_info.id()) {
+                true => Ok(()),
+                false => Err("you do not own this scope"),
+            },
+        },
+    };
```

Comment:
Can we wrap up this code into a method on `WriteAccess`? We should be able to have a method like `get_write_for_package(&self, PackageId)` that gives the user write access to the scope if they should have it, then returns an `anyhow::Result<()>`.